src/lib.rs

```diff
@@ -209,156 +207,58 @@ impl Default for PlaceholderAttrs {
-}
 
-impl Default for CosmicEditUiBundle {
+#[derive(Component)]
+pub struct CosmicCanvas(pub Handle<Image>);
+
+impl Default for CosmicCanvas {
     fn default() -> Self {
-        Self {
-            focus_policy: FocusPolicy::Block,
-            node: Default::default(),
-            button: Default::default(),
-            style: Default::default(),
-            border_color: BorderColor(Color::NONE),
-            interaction: Default::default(),
-            fill_color: Default::default(),
-            image: Default::default(),
-            transform: Default::default(),
-            global_transform: Default::default(),
-            visibility: Default::default(),
-            computed_visibility: Default::default(),
-            z_index: Default::default(),
-            text_position: Default::default(),
-            cosmic_metrics: Default::default(),
-            cosmic_attrs: Default::default(),
-            background_image: Default::default(),
-            max_lines: Default::default(),
-            max_chars: Default::default(),
-            text_setter: Default::default(),
-            mode: Default::default(),
-            background_color: BackgroundColor(Color::WHITE),
-            placeholder_setter: Default::default(),
-            placeholder_attrs: Default::default(),
-        }
+        CosmicCanvas(DEFAULT_IMAGE_HANDLE.typed())
     }
 }
 
-#[derive(Bundle)]
-pub struct CosmicEditSpriteBundle {
-    // Bevy Sprite Bits
-    pub sprite: Sprite,
-    pub transform: Transform,
-    pub global_transform: GlobalTransform,
-    pub texture: Handle<Image>,
-    /// User indication of whether an entity is visible
-    pub visibility: Visibility,
-    /// Algorithmically-computed indication of whether an entity is visible and should be extracted for rendering
-    pub computed_visibility: ComputedVisibility,
-    /// Widget background color
-    pub fill_color: FillColor,
+#[derive(Bundle, Default)]
+pub struct CosmicEditBundle {
     // cosmic bits
-    /// text positioning enum
+    pub fill_color: FillColor,
     pub text_position: CosmicTextPosition,
-    /// text metrics
-    pub cosmic_metrics: CosmicMetrics,
-    /// text attributes
-    pub cosmic_attrs: CosmicAttrs,
-    /// bg img
+    pub metrics: CosmicMetrics,
+    pub attrs: CosmicAttrs,
     pub background_image: CosmicBackground,
-    /// How many lines are allowed in buffer, 0 for no limit
     pub max_lines: CosmicMaxLines,
-    /// How many characters are allowed in buffer, 0 for no limit
     pub max_chars: CosmicMaxChars,
-    /// Setting this will update the buffer's text
     pub text_setter: CosmicText,
-    /// Text input mode
     pub mode: CosmicMode,
-    /// Setting this will update the placeholder text
-    pub placeholder_setter: PlaceholderText,
-    pub placeholder_attrs: PlaceholderAttrs,
+    pub canvas: CosmicCanvas,
 }
 
-impl Default for CosmicEditSpriteBundle {
+#[derive(Bundle)]
+pub struct CosmicEditUiBundle {
```

Comment:
In case of basic sprite example we use bevy's SpriteBundle, can we use NodeBundle or ButtonBundle in basic ui case?

Reply:
Could do, need to provide a UiImage too as that's what is used as canvas for UI elements. I was thinking of it less as a wrapper and more as a convenience thing, adding the defaults with one bundle rather than typing out all the UI components each time

Reply:
Bevy's ButtonBundle already has UiImage, right?

Reply:
Ooh so it does, I'll set this PR to draft and make changes tomorrow. Not sure if the Button tag component will cause any issues with a11y down the line but can cross that bridge when we come to it